From a6692659f565bceffb620145aecf5de85f7f8b4c Mon Sep 17 00:00:00 2001
From: Sylvain Pichon <sylvainx.pichon@intel.com>
Date: Thu, 22 Mar 2012 11:03:33 +0100
Subject: [PATCH 03/49] [PORT FROM MAIN] FM: libhardware_legacy changes for FM
 mode

BZ: 20378

Add FM mode + related functions (setFmRxMode....)
This mode will be used to handled FM RX audio stream.
setFmRxMode function will propagate the FM RX mode (ON/OFF)
from high level framework (appli) to HAL

Change-Id: Ifc2279e15697d8121bd015c221a5d7e955d335b1
Orig-Change-Id: If8ed4b18d41f2f40a54b12e368602ff624cbc1d5
Orig-Change-Id: Ia9e60b369abdd4e77ca5e455df96bf2649b15433
Signed-off-by: Sylvain Pichon <sylvainx.pichon@intel.com>
Reviewed-on: http://android.intel.com:8080/75599
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 audio/audio_hw_hal.cpp |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/audio/audio_hw_hal.cpp b/audio/audio_hw_hal.cpp
index 07d735d..cc81ba0 100644
--- a/audio/audio_hw_hal.cpp
+++ b/audio/audio_hw_hal.cpp
@@ -447,6 +447,12 @@ static int adev_set_mode(struct audio_hw_device *dev, audio_mode_t mode)
     return ladev->hwif->setMode((int) mode);
 }
 
+static int adev_set_fmrx_mode(struct audio_hw_device *dev, int mode)
+{
+    struct legacy_audio_device *ladev = to_ladev(dev);
+    return ladev->hwif->setFmRxMode(mode);
+}
+
 static int adev_set_mic_mute(struct audio_hw_device *dev, bool state)
 {
     struct legacy_audio_device *ladev = to_ladev(dev);
@@ -655,6 +661,7 @@ static int legacy_adev_open(const hw_module_t* module, const char* name,
     ladev->device.set_master_volume = adev_set_master_volume;
     ladev->device.get_master_volume = adev_get_master_volume;
     ladev->device.set_mode = adev_set_mode;
+    ladev->device.set_fmrx_mode = adev_set_fmrx_mode;
     ladev->device.set_mic_mute = adev_set_mic_mute;
     ladev->device.get_mic_mute = adev_get_mic_mute;
     ladev->device.set_parameters = adev_set_parameters;
-- 
1.7.9

